src/batch/src/rpc/service/task_service.rs

```diff
@@ -105,4 +112,47 @@ impl TaskService for BatchServiceImpl {
             }
         }
     }
+
+    #[cfg_attr(coverage, no_coverage)]
+    async fn execute(
+        &self,
+        req: Request<ExecuteRequest>,
+    ) -> Result<Response<Self::ExecuteStream>, Status> {
+        let peer_addr = req
+            .remote_addr()
+            .ok_or_else(|| Status::unavailable("connection unestablished"))?;
+        let req = req.into_inner();
+        let task_id = req.get_task_id().expect("no task id found");
+        let res = self
+            .mgr
+            .fire_task(
```

Comment:
It would be better to skip task manager and construct executor directly here since task manager is used for managing mpp tasks, which will involve monitoring, lifecycle management etc, which would be unnecessary for local execution.